### PR TITLE
parse package-info.class to collect package annotations

### PR DESCRIPTION
**Pull request by ndeloof**
see https://github.com/jenkinsci/lib-annotation-indexer/pull/5

### CODE REVIEW

**Comment by ndeloof** [comment]
maybe it could cache the result for parsing package-info.class as this will be invoked for each and every class in a package. I tried to keep it simple. Performance penalty might not be a big deal as this is a build time tool.

File: access-modifier-annotation/pom.xml [hunk 1]
```diff
@@ -14,7 +14,7 @@
     <dependency>
       <groupId>org.jenkins-ci</groupId>
       <artifactId>annotation-indexer</artifactId>
-      <version>1.4</version>
+      <version>1.13-SNAPSHOT</version> <!-- https://github.com/jenkinsci/lib-annotation-indexer/pull/5 -->
```
Comment (jglick) [inline access-modifier-annotation/pom.xml]: `mvn deploy` to create a timestamped snapshot please.

File: access-modifier-checker/src/main/java/org/kohsuke/accmod/impl/Checker.java [hunk 2]
```diff
@@ -138,85 +138,22 @@ public void loadRestrictions(InputStream stream, final boolean isInTheInspectedM
...
         String className;
         while ((className=r.readLine())!=null) {
-            InputStream is = dependencies.getResourceAsStream(className.replace('.','/') + ".class");
-            if (is==null) {
-                errorListener.onWarning(null,null,"Failed to find class file for "+ className);
-                continue;
-            }
-
-            try {
-                new ClassReader(is).accept(new ClassVisitor(Opcodes.ASM5) {
```
Comment (jglick) [inline access-modifier-checker/src/main/java/org/kohsuke/accmod/impl/Checker.java]: Can you please revert what seems to be an unrelated extract refactoring which makes this very hard to review?

Comment (ndeloof) [reply access-modifier-checker/src/main/java/org/kohsuke/accmod/impl/Checker.java]: not unnecessary, I had to extract a superclass to share code between the class and the package based visitor

Comment (ndeloof) [reply access-modifier-checker/src/main/java/org/kohsuke/accmod/impl/Checker.java]: to give some details, RestrictionsClassVisitor + AnnotatedClassVisitor is == to the initial anonymous class, but this parent class allows to share code with AnnotatedPackageVisitor, which is the sole new code here

**Comment by kohsuke** [comment]
For a small library like this I'm not too hang up on "revert unrelated refactoring" but just as a note to myself, this PR is still blocked on the upstream release.

**Comment by ndeloof** [comment]
inactive, feel free to take over
